src/sentry/api/endpoints/organization_stats_v2.py

```diff
@@ -0,0 +1,239 @@
+from sentry.utils.outcomes import Outcome
+
+from sentry.utils.snuba import (
+    naiveify_datetime,
+    to_naive_timestamp,
+)
+
+
+CATEGORY_NAME_MAP = {
+    DataCategory.ERROR: "statsErrors",
+    DataCategory.TRANSACTION: "statsTransactions",
+    DataCategory.ATTACHMENT: "statsAttachments",
+}
+
+
+class OrganizationStatsEndpointV2(OrganizationEndpoint):
+    def get(self, request, organization):
+        start, end, rollup = get_date_range_rollup_from_params(request.GET, "1h", round_range=True)
+        result = outcomes.query(
+            start=start,
+            end=end,
+            rollup=rollup,
+            groupby=["category", "time", "outcome", "reason"],
+            aggregations=[["sum", "times_seen", "times_seen"], ["sum", "quantity", "quantity"]],
+            filter_keys={"org_id": [organization.id]},
+            orderby=["time"],
+        )
+
+        response = StatsResponse(start, end, rollup)
+        for row in result:
+            stat_to_update = response.get(row["category"])
+            stat_to_update.update(row)
+
+        # response.zerofill(start, end, rollup)
+        return Response(response.build_fields())
+
+
+class OrganizationProjectStatsIndex(OrganizationEndpoint):
+    def get(self, request, organization):
+        start, end, rollup = get_date_range_rollup_from_params(request.GET, "1h", round_range=True)
+
+        # TODO: see if there's a better way to get the user's projects
+        project_list = []
+        team_list = Team.objects.get_for_user(organization=organization, user=request.user)
+        for team in team_list:
+            project_list.extend(Project.objects.get_for_user(team=team, user=request.user))
+        project_ids = list({p.id for p in project_list})
+
+        result = outcomes.query(
+            start=start,
+            end=end,
+            rollup=rollup,
+            groupby=["project_id", "category", "time", "outcome", "reason"],
+            aggregations=[["sum", "times_seen", "times_seen"], ["sum", "quantity", "quantity"]],
+            filter_keys={"org_id": [organization.id], "project_id": project_ids},
+            orderby=["times_seen", "time"],
+        )
+
+        response = {project_id: StatsResponse(start, end, rollup) for project_id in project_ids}
+        for row in result:
+            stat_to_update = response[row["project_id"]].get(row["category"])
+            stat_to_update.update(row)
+
+        return Response(
+            {project_id: stat_res.build_fields() for project_id, stat_res in response.items()}
+        )
+
+
+class OrganizationProjectStatsDetails(ProjectEndpoint, OrganizationEndpoint):
+    def get(self, request, project):
+        start, end, rollup = get_date_range_rollup_from_params(request.GET, "1h", round_range=True)
+
+        # TODO: see if there's a better way to get the user's projects
+        project_list = []
+        team_list = Team.objects.get_for_user(organization=project.organization, user=request.user)
```

Comment:
You can/should use `OrganizationEndpoint.get_projects()` for this. It handles both kinds of 'all' and organization permissions around open team membership which would be needed here too.

Reply:
👍🏼